vortex-array/src/array/sparse/canonical.rs

```diff
@@ -0,0 +1,213 @@
+use arrow_buffer::BooleanBufferBuilder;
+use vortex_dtype::{match_each_native_ptype, DType, NativePType};
+use vortex_error::{VortexError, VortexResult};
+use vortex_scalar::ScalarValue;
+
+use crate::array::primitive::PrimitiveArray;
+use crate::array::sparse::SparseArray;
+use crate::array::BoolArray;
+use crate::validity::Validity;
+use crate::{ArrayDType, Canonical, IntoArrayVariant, IntoCanonical};
+
+impl IntoCanonical for SparseArray {
+    fn into_canonical(self) -> VortexResult<Canonical> {
+        // Resolve our indices into a vector of usize, applying the offset
+        let indices = self.resolved_indices();
+
+        if matches!(self.dtype(), DType::Bool(_)) {
+            canonicalize_sparse_bools(
+                self.values().into_bool()?,
+                &indices,
+                self.len(),
+                self.fill_value(),
+            )
+        } else {
+            let values = self.values().into_primitive()?;
+            match_each_native_ptype!(values.ptype(), |$P| {
+                canonicalize_sparse_primitives(
+                    values.maybe_null_slice::<$P>(),
+                    values.validity(),
+                    &indices,
+                    self.len(),
+                    self.fill_value()
+                )
+            })
+        }
+    }
+}
+
+fn canonicalize_sparse_bools(
+    values_array: BoolArray,
+    indices: &[usize],
+    len: usize,
+    fill_value: &ScalarValue,
+) -> VortexResult<Canonical> {
+    let fill_bool: bool = if fill_value.is_null() {
+        bool::default()
```

Comment:
nit - a `false` here is much easier to read IMO, also doesn't need the type hint in line 45